[DirectoryFactory] Store CFileItem mimetype to CURL, to avoid forced HTTP HEAD requests

CastagnaIT opened the pull request:
## Description
<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here. -->
This will fix inputstream PVR binary addons and playlists like STRM that set mimetype to the CFileItem or by using `#KODIPROP:mimetype`

currently the mimetype is passed correctly to Kodi core,
but when the FileItem arrive to `CDirectoryFactory::Create` is converted to `CURL` object
which leads to the loss of all CFileItem infos
then at later time a new `CFileItem` will be created based on the `CURL` object, since there is no CFileItem original infos, is created from scratch, without mimetype

After investigated more, i found that  `CURL` object can host optional extradata, so i stored the mimetype to it, to avoid deeper changes

the `PLAYLIST::CPlayListFactory::Create` with CURL, is used only on the methods of `CPlaylistFileDirectory` seems to be quite reliable as change

~_i havent had time to check if it is needed backport to kodi21, i will check at later time or tomorrow_~ backport needed

## Motivation and context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
fix #25652
## How has this been tested?
<!--- Please describe in detail how you tested your change. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Can be tested with these fake STRM (PVR use same code path)
Multi videos playlist: [MULTI_FAKE.strm.txt](https://github.com/user-attachments/files/16740481/MULTI_FAKE.strm.txt)
Single video: [SINGLE_FAKE.strm.txt](https://github.com/user-attachments/files/16740482/SINGLE_FAKE.strm.txt)

Videos have fake links, just to check if on the log debug you see printed `CCurlFile::XFILE::CCurlFile::GetMimeType` errors
that should not appears if the fix works

other way to test it is use PVR pvr.iptvsimple by using appropriate STRM's

## What is the effect on users?
<!--- Summarize the effect of this change on Kodi end-users. -->
<!--- If the PR does not have a noticeable impact (e.g., if it only changes documentation), -->
<!--- just leave it empty. Put in more detail the bigger the impact is. -->
<!--- This section may be used for automatic creation of release notes. -->
Users can have `#KODIPROP:mimetype` working on playlists (currently broken)
PVR addons can works without that kodi core force make HTTP HEAD requests (so without take in account of the specified mimetype)

some particular video services dont like HTTP HEAD requests and can lead to broken playback beause block manifest download (this is caused often due to custom kodi user-agent), this will prevent this problem

On windows noticieable kodi freezes for some secs happens each time HTTP HEAD request fails, this seem to be a CURL problem, anyway this allow to avoid also this problem

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` with no space in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

ksooo commented:
There are other places where `CPlayList* CPlayListFactory::Create(const std::string& filename)` is called. Are they all uneffected from the bug you are fixing here? Or should they also be fixed?

<img width="381" alt="image" src="https://github.com/user-attachments/assets/705faf08-6f84-440e-9e3c-61f8f4ced669">

CastagnaIT commented:
i will check, atm my computer is burned so i cant check, i need to buy new one

basilgello commented:
@CastagnaIT How comes? Electricity drop and then surge?

CastagnaIT commented:
yes here i have unstable electricity when rains heavily, a power failure of about 1 second happened
too fast, all household appliances turned off, but the pc suffered from the short absence but did not turn off,
monitor turned black screen and full dead motherboard and cpu, i hope the hdd is ok but i dont have an adapter to test it

basilgello commented:
Oh… I had such crap in Ukraine ~12 years ago, ended up buying 4 UPSes and replacing batteries only once in 10 years in each…

CastagnaIT committed:
[DirectoryFactory] If FileItem has mimetype store to CURL

This to be restored when the FileItem will be reconstructed later
to prevent make HTTP HEAD requests

CastagnaIT commented:
@ksooo i have investigated
all mentioned methods are not affected for playlists cases,
because the playlist data is loaded later time (after make make the factory)

some others remarks for these (that at the end are not related):
CPlaylistFileDirectory::ContainsFiles: i have not found uses

CPlaylistFileDirectory::GetDirectory: there is only another use of this method in the  CGUIWindowMusicPlaylistEditor::LoadPlaylist but the use case is different/not related since there is no CLIstItem

For CFileDirectoryFactory::Create, all other uses of this method are not related to playlists

XSPF body test, not related

just rebased PR and add a little code comment

CastagnaIT commented:
@ksooo  we can go on also with this PR? at least if it will be merged you can have feedbacks in case there are special use cases that are not immediately detectable
